sw/device/lib/crypto/drivers/entropy.c

```diff
@@ -361,10 +390,11 @@ static status_t csrng_send_app_cmd(uint32_t reg_address,
     return OTCRYPTO_RECOV_ERR;
   }
 
-  if (check_completion) {
+  if (check_completion && (cmd_type == kEntropyCsrngSendAppCmdTypeCsrng)) {
```

Comment:
In this and the following check, is `check_completion` actually serving a purpose now? Should we just remove that argument from the function and use `cmd_type` instead to determine where and whether to look for the done bit?

Reply:
I guess this could still have a use case if one would like to send a CSRNG/EDN_SW command without waiting for the acknowledgement.